[EDU-6180] Edge Function template guide (#1488)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/templates/edge-function-starter-kit.mdx

---
title: How to deploy the Edge Function Starter Kit template
description: The Edge Function Starter Kit template allows you to create an edge function-based application without origin dependencies. 

> meta_tags: templates, guides, Azion Marketplace, web, Edge Function, Edge Application

Reviewer: Add "starter kit"?

namespace: docs_guides_edge_function_starter_kit
permalink: /documentation/products/guides/edge-function-starter-kit/
---

import Tag from 'primevue/tag';

<Tag severity="info" client:only="vue">
Preview
</Tag>

> The **Edge Function Starter Kit** template contains the configurations for creating an edge function-based application without origin dependencies. This template simplifies the configuration process, allowing you to quickly deploy an edge function that is ready to run through the Azion Edge Platform. 

Reviewer 2: ```suggestion
The **Edge Function Starter Kit** template contains the configurations for creating an edge function-based application without origin dependencies. This template simplifies the configuration process, allowing you to quickly deploy an edge function that is ready to run at the edge. 
```

The deployment of this template creates an edge application, a basic "Hello World” edge function, and a domain to facilitate your access and management through Azion Edge Platform.

> 
> 
> ## Requirements

Reviewer 2: ```suggestion

---

## Requirements
```

2. Click the **Deploy** button to start the deployment process.

During the deployment, you'll be able to follow the process through a window showing off the logs. When it's complete, the page shows information about the application and some options to continue your journey.

:::note
The link to the edge application allows you to see it on the browser. However, it takes a certain time to propagate and configure the application in Azion's edge locations. It may be necessary to wait a few minutes for the URL to be activated and for the application page to be effectively displayed in the browser.
:::

---

## Managing the template

Considering that this initial setup may not be optimal for your specific edge application, all settings can be customized any time you need by using Azion's platform.

### Updating the edge function

This template includes a simple `Hello World` edge function that is ready to run. Follow these steps to update it according to your needs:

1. [Access Azion Console](https://console.azion.com).
2. On the upper-left corner, select **Products menu**, the icon with three horizontal lines, > **Edge Functions**.

> 3. Find the edge function related to your template and select it. It will have the same name of your edge application.

Reviewer 2: ```suggestion
3. Find the edge function related to your template and select it. It'll have the same name as your edge application.
```

1. [Access Azion Console](https://console.azion.com).
2. On the upper-left corner, select **Products menu**, the icon with three horizontal lines, > **Edge Application**.
    - You'll be redirected to the **Edge Application** page. It lists all the edge applications you've created.
3. Find the edge application related to your template and select it.
    - The list is organized alphabetically. You can also use the **search bar** located in the upper-left corner of the list; currently, it filters only by **Application Name**.

After selecting the edge application you'll work on, you'll be directed to a page containing all the settings you can configure.

:::tip
Read the documentation about [managing edge applications](/en/documentation/products/build/edge-application/first-steps/) for more details.
:::

### Adding a custom domain

The edge application created during the deployment has an assigned Azion domain to make it accessible through the browser. The domain has the following format: `xxxxxxxxxx.map.azionedge.net/`. However, you can add a custom domain for users to access your edge application through it.

import LinkButton from 'azion-webkit/linkbutton';

> <LinkButton link="/en/documentation/products/guides/configure-a-domain/" label="go to configuring a domain guide" severity="secondary" /> 

Reviewer: As per button on "https://www.azion.com/en/documentation/products/guides/cosmic-agency-website/"
```suggestion
<LinkButton link="/en/documentation/products/guides/configure-a-domain/" label="Go to Configuring a Domain Guide" severity="secondary" /> 
```
